src/EntityManagerFactory.php

```diff
@@ -441,4 +445,44 @@ protected function registerMappingTypes(array $settings = [], EntityManagerInter
             $manager->getConnection()->getDatabasePlatform()->registerDoctrineTypeMapping($dbType, $doctrineType);
         }
     }
+
+    /**
+     * Check if master slave connection was being configured.
+     *
+     * @param array $driverConfig
+     *
+     * @return bool
+     */
+    private function isMasterSlaveConfigured(array $driverConfig)
+    {
+        // Setting read is mandatory for master/slave configuration. Setting write is optional.
+        // But if write was set and read wasn't, it means configuration is incorrect and we must inform the user.
+        return isset($driverConfig['read']) || isset($driverConfig['write']);
+    }
+
+    /**
+     * Check if slave configuration is valid.
+     *
+     * @param array $driverConfig
+     *
+     * @return bool
+     */
+    private function hasValidMasterSlaveConfig(array $driverConfig)
+    {
+        if (!isset($driverConfig['read'])) {
+            throw new \InvalidArgumentException("Parameter 'read' must be set for read/write config.");
+        }
+
+        $slaves = $driverConfig['read'];
+
+        if (!is_array($slaves) || in_array(false, array_map('is_array', $slaves))) {
+            throw new \InvalidArgumentException("Parameter 'read' must be an array containing multiple arrays.");
+        }
+
+        if (($key = array_search(0, array_map('count', $slaves))) && $key !== false) {
```

Comment:
isn't `$key !== false` redundant here? if its `false`, first condition fails.
Search could return position 0, which would be false-_ish_ here as well, right?